lib/Sema/TypeCheckNameLookup.cpp

```diff
@@ -273,6 +273,25 @@ LookupResult TypeChecker::lookupUnqualified(DeclContext *dc, DeclNameRef name,
   return result;
 }
 
+// An unfortunate hack to kick the decl checker into adding semantic members to
+// the current type before we attempt a semantic lookup. The places this method
+// looks needs to be in sync with \c extractDirectlyReferencedNominalTypes.
+// See the note in \c synthesizeSemanticMembersIfNeeded about a better, more
+// just, and peaceful world.
+static void installSemanticMembersIfNeeded(Type type, DeclNameRef name) {
+  // Look-through class-bound archetypes to ensure we synthesize e.g.
+  // inherited constructors.
+  if (auto archetypeTy = type->getAs<ArchetypeType>()) {
```

Comment:
Can you also add a test for a superclass-constrained existential `C & P` and add a check here if needed?

```
if (type->isExistentialType()) {
  auto layout = type.getExistentialLayout();
  if (auto superclass = layout.explicitSuperclass) {
```

Finally, we should probably handle lookups on a metatype as well, although I don't remember if we have any static semantic members so maybe you can't hit the problem in practice yet. That can be handled by unwrapping an AnyMetatypeType before doing the other checks.

Reply:
Never mind, I think metatypes are unwrapped further up the stack, and then callers filter out static or instance members as needed. The existential case should probably be handled though.